apps/public/public.go

```diff
@@ -522,13 +522,59 @@ type interactiveAuthOptions struct {
 	claims, domainHint, loginHint, redirectURI, tenantID string
 	openURL                                              func(url string) error
 	authnScheme                                          AuthenticationScheme
+	successPage                                          []byte
+	errorPage                                            []byte
 }
 
 // AcquireInteractiveOption is implemented by options for AcquireTokenInteractive
 type AcquireInteractiveOption interface {
 	acquireInteractiveOption()
 }
 
+func WithSuccessPage(successPage []byte) interface {
```

Comment:
From the public API standpoint, I think we'd want an API more like `WithSystemBrowserOptions(succesPage, errorPage)`. This is because we may introduce support for broker, these options won't make sense .

CC @localden

Reply:
Agree - this needs to be mapped to the API concepts from other MSALs. Additionally, would be good to get @rayluo's input here - we _likely_ need to find a way to sanitize inputs here so that we don't end up in a scenario where someone can pass in malicious code into the rendered page.

Reply:
@bgavrilMS @chlowell - just a sanity check on my part, do we need some sanitization logic here, similar to what we've updated in Python?

Reply:
I can imagine a scenario in which an attacker steals a token by modifying an app's success page but whether that's something MSAL must (try to) prevent depends on the threat model.

Reply:
Hi @chlowell , Can you please explain what to check for here?  (_my setup is not ideal yet, and I was not able to inspect the http headers to see if the token is present, when serving the error / success page_)

Reply:
Is there a preferred / accepted library for sanitising HTML ?

I found a few libraries ([this one ](https://pkg.go.dev/github.com/microcosm-cc/bluemonday#section-readme)stands out a bit more than others) that could help with sanitising HTML.

Reply:
> I can imagine a scenario in which an attacker steals a token by modifying an app's success page but whether that's something MSAL must (try to) prevent depends on the threat model.

I thought I wrote a reply for this message but apparently it did not go through. What I tried to say was that, a success page typically contains only a hardcoded message such as "sign-in succeeded, you can close this tab now" so it does not have XSS risk. It is the error page that acts as a template containing variable(s) for error content, which we need sanitising before rendering.